app-connector-client/server/routes/localApis.js

```diff
@@ -14,23 +14,50 @@ function getAll(req, res) {
     var configApis = varkesConfig.apis;
     for (var i = 0; i < configApis.length; i++) {
         var api = configApis[i]
-
-        apis.push(services.fillServiceMetadata(api, req.body.hostname || req.headers.host))
+        let metadata = services.fillServiceMetadata(api, req.body.hostname || req.headers.host)
+        metadata.id = api.name;
+        apis.push(metadata)
     }
     var configEvents = varkesConfig.events;
     for (var i = 0; i < configEvents.length; i++) {
         var event = configEvents[i];
-        apis.push(services.fillEventData(event));
+        let metadata = services.fillEventData(event)
+        metadata.id = event.name;
+        apis.push(metadata);
     }
     res.status(200).send(apis);
 }
+function getLocalApi(req, res) {
+    LOGGER.debug("Getting Local API")
+    let apiname = req.params.apiname;
+    let api = varkesConfig.apis.find(x => x.name == apiname);
+    if (api) {
+        let serviceMetadata = services.fillServiceMetadata(api, req.headers.host)
+        serviceMetadata.id = apiname;
+        res.status(200).send(serviceMetadata);
+    }
+    else {
+        api = varkesConfig.events.find(x => x.name == apiname);
+        if (api) {
+            let eventMetadata = services.fillEventData(api)
+            eventMetadata.id = apiname;
+            res.status(200).send(eventMetadata);
+        }
+        else {
+            let message = "api " + apiname + " does not exist";
+            LOGGER.error(message);
+            res.status(404).send({ error: message })
+        }
+    }
+}
 async function registerAll(req, res) {
     LOGGER.debug("Registering all Local APIs")
     var err = assureConnected()
     if (err) {
         res.status(400).send({ error: err })
     }
     try {
+        services.initBatchRegistration();

```

Comment:
As discussed, please refactor, so that you call just one method which init and start registering all.

Reply:
done